File: android/versioned-abis/expoview-abi32_0_0/src/main/java/abi32_0_0/expo/modules/filesystem/FileSystemPackage.java

package abi32_0_0.expo.modules.filesystem;

import android.content.Context;

import java.util.Collections;
import java.util.List;

import abi32_0_0.expo.core.ExportedModule;
import abi32_0_0.expo.core.BasePackage;
import abi32_0_0.expo.core.interfaces.InternalModule;

public class FileSystemPackage extends BasePackage {
  @Override
  public List<InternalModule> createInternalModules(Context context) {
    return Collections.<InternalModule>singletonList(new FilePermissionModule());
  }

  @Override
  public List<ExportedModule> createExportedModules(Context context) {
    return Collections.<ExportedModule>singletonList(new FileSystemModule(context));
  }
}

File: android/versioned-abis/expoview-abi46_0_0/src/main/java/abi46_0_0/expo/modules/filesystem/FileSystemPackage.kt
package abi46_0_0.expo.modules.filesystem

import android.content.Context
import abi46_0_0.expo.modules.core.BasePackage
import abi46_0_0.expo.modules.core.ExportedModule
import abi46_0_0.expo.modules.core.interfaces.InternalModule

class FileSystemPackage : BasePackage() {
  override fun createInternalModules(context: Context): List<InternalModule> =
    listOf<InternalModule>(FilePermissionModule())

  override fun createExportedModules(context: Context): List<ExportedModule> =
    listOf<ExportedModule>(FileSystemModule(context))
}
